scripts/map_tokens.py

```diff
@@ -0,0 +1,14 @@
+from datasets import DatasetDict, IterableDataset, IterableDatasetDict
+from datasets.arrow_dataset import Dataset
+
+
+def token_map(
+    tokenized_dataset: DatasetDict | Dataset | IterableDatasetDict | IterableDataset,
```

Comment:
why so many?

Reply:
You can try using https://docs.python.org/3/library/typing.html#typing.cast let pyright / pylance know we will be only using Dataset only before calling this function